File: 2022/day2.py
# Opponent - A for Rock, B for Paper, and C for Scissors
# Response - X for Rock, Y for Paper, and Z for Scissors

h0 = open("day2.txt")
l0 = h0.readlines()[:-1]
l0 = [el.strip().split(" ") for el in l0]

def move_points(p0):
    rules = {"X": 1, "Y": 2, "Z": 3}
    return rules[p0]

def win_points(p0, p1):
    rules ={ ("X","C"): 6, ("Y","A"): 6, ("Z","B"): 6, 
             ("X","A"): 3, ("Y","B"): 3, ("Z","C"): 3,
             ("X","B"): 0, ("Y","C"): 0, ("Z","A"): 0
           }
    points = rules[(p0,p1)]
    return points

points = sum([win_points(round[1],round[0]) + move_points(round[1]) for round in l0])
print(points)

    

File: 2022/day2.rs
use std::collections::HashMap;
use std::io::{self, BufReader};
use std::io::prelude::*;
use std::fs::File;

fn main() -> io::Result<()> {
    //let f = File::open("day2_ex.txt")?;
    let f = File::open("day2.txt")?;
    let f = BufReader::new(f);
    
    let mut vec = Vec::new();

    for line in f.lines() {
        let line = line.unwrap();
        //println!("{}", line);
        vec.push(line);
    }
    
    let mut total: i32 = 0;
    for pair in vec {
        let split = pair.split(" ");
        let vp = split.collect::<Vec<&str>>();
        total += move_points(vp[1]);
        total += win_points(vp[0], vp[1]);
    }

    println!("{:?}", total);
    Ok(())
}

fn move_points(p0: &str) -> i32 {
    let rules = HashMap::from([
        ("X", 1), ("Y", 2), ("Z", 3),
        ]);
    //println!("{}", p0);
    rules[&p0]
}
fn win_points(p0: &str, p1: &str) -> i32 {
    let rules = HashMap::from([
             (("X","C"), 6), (("Y","A"), 6), (("Z","B"), 6), 
             (("X","A"), 3), (("Y","B"), 3), (("Z","C"), 3),
             (("X","B"), 0), (("Y","C"), 0), (("Z","A"), 0),
        ]);
    let key = (p1, p0);
    //println!("{:?}", key);
    rules[&key]
}
